### DISCUSSION TITLE
Data package with golem and Shiny

### DISCUSSION THREAD

**Discussion by ShinyFabio** [ThinkR-open/golem#669] [Q&A]
Hello,
I'm developing a shiny app with golem. This app requires a lot of photos that are inside the www folder (with a lot of subfolders required by the app in order to show only some photos) and so the package is a bit heavy. Since these photos are not mandatory (the app works without them) I was thinking about create a data package that can be installed later if the user need them. Do you have any idea how to make it? Long story short, I need a data package that put all my photos in the www folder of my app, mantaining the subfolders structure.

**Comment by ColinFay** [comment]
Hey! 

So, this part of the `app_ui` allows to link any folder of your server as a resource path for your app 

https://github.com/ThinkR-open/golem/blob/master/inst/shinyexample/R/app_ui.R#L28

Here, you could specify any other path, for example with something like: 

```r
add_resource_path(
	'img', system.files('img', package = 'myotherpackage')
)
```

That way, you can link the folder from the other package. 

If you want to link these only if the other package is installed, I'd suggest doing something like: 

```r
if (requireNamespace("myotherpackage"){
	add_resource_path(
		'img', system.files('img', package = 'myotherpackage')
	)
} 
```

Let me know if that works, 

Colin

**Comment by ShinyFabio** [comment]
Thank you! It looks perfect. I'm going to try your solution. I'll let you know later!
